packages/react/src/runtime/__tests__/ax.test.ts

```diff
@@ -1,70 +1,81 @@
 import ax from '../ax';
 
 describe('ax', () => {
-  it('should join single classes together', () => {
-    const result = ax(['foo', 'bar']);
-
-    expect(result).toEqual('foo bar');
-  });
-
-  it('should join multi classes together', () => {
-    const result = ax(['foo baz', 'bar']);
-
-    expect(result).toEqual('foo baz bar');
-  });
-
-  it('should remove undefined', () => {
-    const result = ax(['foo', 'bar', undefined]);
-
-    expect(result).toEqual('foo bar');
-  });
-
-  it('should ensure the last atomic declaration of a single group wins', () => {
-    const result = ax(['_aaaabbbb', '_aaaacccc']);
-
-    expect(result).toEqual('_aaaacccc');
-  });
-
-  it('should ensure the last atomic declaration of many single groups wins', () => {
-    const result = ax(['_aaaabbbb', '_aaaacccc', '_aaaadddd', '_aaaaeeee']);
-
-    expect(result).toEqual('_aaaaeeee');
-  });
-
-  it('should ensure the last atomic declaration of a multi group wins', () => {
-    const result = ax(['_aaaabbbb _aaaacccc']);
-
-    expect(result).toEqual('_aaaacccc');
-  });
-
-  it('should ensure the last atomic declaration of many multi groups wins', () => {
-    const result = ax(['_aaaabbbb _aaaacccc _aaaadddd _aaaaeeee']);
-
-    expect(result).toEqual('_aaaaeeee');
-  });
-
-  it('should not remove any atomic declarations if there are no duplicate groups', () => {
-    const result = ax(['_aaaabbbb', '_bbbbcccc']);
-
-    expect(result).toEqual('_aaaabbbb _bbbbcccc');
-  });
-
-  it('should not apply conditional class', () => {
-    const isEnabled: boolean = (() => false)();
-    const result = ax([isEnabled && 'foo', 'bar']);
-
-    expect(result).toEqual('bar');
-  });
-
-  it('should ignore non atomic declarations', () => {
-    const result = ax(['hello_there', 'hello_world']);
-
-    expect(result).toEqual('hello_there hello_world');
-  });
-
-  it('should ignore non atomic declarations when atomic declarations exist', () => {
-    const result = ax(['hello_there', 'hello_world', '_aaaabbbb']);
-
-    expect(result).toEqual('hello_there hello_world _aaaabbbb');
+  const isEnabled: boolean = (() => false)();
+
+  it.each([
+    ['should handle empty array', [], undefined],
+    ['should handle array with undefined', [undefined], undefined],
+    ['should join single classes together', ['foo', 'bar'], 'foo bar'],
+    ['should join multi classes together', ['foo baz', 'bar'], 'foo baz bar'],
+    ['should remove undefined', ['foo', 'bar', undefined], 'foo bar'],
+    [
+      'should ensure the last atomic declaration of a single group wins',
+      ['_aaaabbbb', '_aaaacccc'],
+      '_aaaacccc',
+    ],
+    [
+      'should ensure the last atomic declaration of a single group with short class name wins',
```

Comment:
I haven't followed up the previous PRs and I am still wrapping my head around these changes. Hence, apologies if this was mentioned somewhere else

I am assuming that:
- `_aaaabbbb` is the CSS class with no manipulation
- `a` is the optimised class, which is stored in the map
- `_aaaa_a` is made of atomic group + optimised class -> in this way we are able to handle overrides

Reply:
Yes, spot on!